### PR TITLE
Update dart.yml

### PR DESCRIPTION
**Pull request by devoncarew**
This PR updates the github actions starter workflow for Dart.

Note that I am a member of the Dart team and am more than happy to answer any questions about these changes.

This PR:
- discusses the other release channels we have available (dev and beta, in addition to the existing `google/dart:latest` stable channel docker image referenced below), as well as includes a link where users can find the list of docker images
- adds a code analysis step
- adds a format verification step
- the format step is commented out as that's more likely to depend on the workflow of specific teams (but useful to know the command line to run)
- updates the test step to use our newer `dart test` tooling (instead of the existing `pub run test` command)
- for the test step, mention the slightly different command that a Flutter project will need to use

Thanks!

---

**Please note that we are not accepting new starter workflows at this time. Updates to existing starter workflows are fine.**

---

In the workflow and properties files:

- [ ] The workflow filename of CI workflows should be the name of the language or platform, in lower case.  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").

  The workflow filename of publishing workflows should be the name of the language or platform, in lower case, followed by "-publish".
- [ ] Includes a matching `ci/properties/*.properties.json` file.
- [ ] Use sentence case for the names of workflows and steps, for example "Run tests".
- [ ] The name of CI workflows should only be the name of the language or platform: for example "Go" (not "Go CI" or "Go Build")
- [ ] Include comments in the workflow for any parts that are not obvious or could use clarification.
- [ ] CI workflows should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.

Some general notes:

- [ ] This workflow must only use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [ ] This workflow must only use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  Workflows using these actions must reference the action using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [ ] This workflow must not send data to any 3rd party service except for the purposes of installing dependencies.
- [ ] This workflow must not use a paid service or product.

### CODE REVIEW

File: ci/dart.yml [hunk 1]
```diff
@@ -8,15 +8,30 @@ on:
...
     steps:
-    - uses: actions/checkout@v2
-    - name: Install dependencies
-      run: pub get
-    - name: Run tests
-      run: pub run test
+      - uses: actions/checkout@v2
+
+      - name: Install package dependencies
+        run: pub get
```
Comment (mit-mit) [inline ci/dart.yml:24]: `dart pub get`

File: ci/dart.yml [hunk 2]
```diff
@@ -8,15 +8,30 @@ on:
...
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Install dependencies
-      run: pub get
-    - name: Run tests
-      run: pub run test
+      - uses: actions/checkout@v2
+
+      - name: Install package dependencies
```
Comment (mit-mit) [inline ci/dart.yml:23]: I've found it useful to add this step at the first step:

```
      - name: Report version
        run: dart --version
```

File: ci/dart.yml [hunk 3]
```diff
@@ -8,15 +8,30 @@ on:
...
+        run: pub get
+
+      # Comment this step in to verify the use of 'dart format' on each commit.
```
Comment (mit-mit) [inline ci/dart.yml:26]: "Comment ... in" is a bit confusing; consider "# Uncomment ..."

**Review by mit-mit — CHANGES_REQUESTED** [review]

File: ci/dart.yml (hunk 3 above)
Comment (devoncarew) [reply ci/dart.yml]: Changed

File: ci/dart.yml (hunk 1 above)
Comment (devoncarew) [reply ci/dart.yml]: updated

File: ci/dart.yml (hunk 2 above)
Comment (devoncarew) [reply ci/dart.yml]: added

**Review by devoncarew — COMMENTED** [review]
Thanks for the review! Updated.

**Comment by mit-mit** [comment]
Great. Why the change to make all the step names are the same as the commands?

**Comment by devoncarew** [comment]
> Great. Why the change to make all the step names are the same as the commands?

When I've done this on CIs (https://github.com/dart-lang/usage/runs/1334831476?check_suite_focus=true, ...) I think it reads a bit better.

**Comment by mit-mit** [comment]
Hmm, it seems a bit inconsistent with "Setup job" and "Initialize containers"?

**Comment by devoncarew** [comment]
> Hmm, it seems a bit inconsistent with "Setup job" and "Initialize containers"?

Yes, likely, but I think it helps clarify to Dart users what the various tasks are doing (there will be less guessing about what's in the 'pub get' or 'dart analyze' sections than there would be for 'Install package dependencies' or 'Analyze project' sections).

**Review by andymckay — COMMENTED** [review]
This looks fine to me, but I agree with name changes. Looking across some of the other starter workflows in this repository, this is not a pattern that's common. Let's keep it the same as the rest with a generic description of what the step is about.

**Comment by devoncarew** [comment]
> Let's keep it the same as the rest with a generic description of what the step is about.

Sure, happy to revert the name changes. I'll update the names and ping this PR again.

**Review by devoncarew — COMMENTED** [review]
PTAL - I revered the step names, and used sentence case for the new step names (`Verify formatting`, ...).

**Review by andymckay — APPROVED** [review]

**Comment by andymckay** [comment]
Thanks for the udpates @mit-mit 🙇🏾

**Pull request merged**